### PR TITLE
Upstream merge up to September

### PR DESCRIPTION
**Pull request by carterqw2**
### Description
Merges upstream commits up to 085bf68427f56498978c81b1aaa333776ea1b2cb

### Issues
Relates to https://github.com/celo-org/data-services/issues/557

### CODE REVIEW

File: apps/block_scout_web/assets/js/lib/csv_download.js [hunk 1]
```diff
@@ -28,39 +29,35 @@ const _instance2 = new Pikaday({
...
-      },
-      data: {
-        type: 'JSON',
-        captchaResponse: resp
+  const recaptchaResponse = grecaptcha.enterprise.getResponse()
+  if (recaptchaResponse) {
+    $button.addClass('spinner')
+    $button.prop('disabled', true)
+    $message.removeClass('hidden')
+    const downloadUrl = `${$button.data('link')}&recaptcha_response=${recaptchaResponse}`
```
Comment (rkachowski) [inline apps/block_scout_web/assets/js/lib/csv_download.js:38]: have we tested this on our branch? IIRC there are significant api differences between recaptcha enterprise (what we use) vs upstream recaptcha

File: apps/explorer/test/explorer/chain/csv_export/address_internal_transaction_csv_exporter_test.exs [hunk 2]
```diff
@@ -91,24 +91,24 @@ defmodule Explorer.Chain.AddressInternalTransactionCsvExporterTest do
           }
         end)
 
-      assert result.transaction_hash == to_string(internal_transaction.transaction_hash)
-      assert result.index == to_string(internal_transaction.index)
-      assert result.block_number == to_string(internal_transaction.block_number)
-      assert result.block_index == to_string(internal_transaction.block_index)
-      assert result.block_hash == to_string(internal_transaction.block_hash)
-      assert result.transaction_index == to_string(internal_transaction.transaction_index)
+      assert result.transaction_hash == [[], to_string(internal_transaction.transaction_hash)]
```
Comment (rkachowski) [inline apps/explorer/test/explorer/chain/csv_export/address_internal_transaction_csv_exporter_test.exs:94]: why is everything now prepended with empty lists?

**Review by rkachowski — APPROVED** [review]
Looks mostly ok. There are some wtf moments in the patches and I am unsure why the recaptcha enterprise verificiation is working against non enterprise endpoints.

There are also some warnings from the new CodeQL check system added to the CI run which I presume will either be fixed or ignored in future upstream merges.

File: apps/explorer/test/explorer/chain/csv_export/address_internal_transaction_csv_exporter_test.exs (hunk 2 above)
Comment (carterqw2) [reply apps/explorer/test/explorer/chain/csv_export/address_internal_transaction_csv_exporter_test.exs:94]: Yeah, it's introduced [here](https://github.com/blockscout/blockscout/commit/f44e4e2cc44bc99d4d129b72267d6fbee58c53af), I think it's a change in the nimble_csv lib.

File: apps/block_scout_web/assets/js/lib/csv_download.js (hunk 1 above)
Comment (carterqw2) [reply apps/block_scout_web/assets/js/lib/csv_download.js:38]: I tested transactions and epoch transactions downloads on staging.

**Review by shazarre — APPROVED** [review]

**Pull request merged**